ckan/logic/action/update.py

```diff
@@ -308,9 +308,11 @@ def package_update(context, data_dict):
         rev.message = _(u'REST API: Update object %s') % data.get("name")
 
     #avoid revisioning by updating directly
-    model.Session.query(model.Package).filter_by(id=pkg.id).update(
-        {"metadata_modified": datetime.datetime.utcnow()})
-    model.Session.refresh(pkg)
+    now = datetime.datetime.utcnow()
+    if not pkg.metadata_modified or pkg.metadata_modified + datetime.timedelta(seconds=10) < now:
+        model.Session.query(model.Package).filter_by(id=pkg.id).update(
```

Comment:
could this be added to the filter_by where it finds zero records or do we already have the data available?

can you add an extra comment why we did this change.

might also be better for 2 seconds, 10seconds may be too large since we are talking about ms lock contention not multi second.

Reply:
Testing with curl suggests that we already have the timestamp loaded at this point. Moving the check into the filter would probably mean that we would still have a database lock even if we don't need one.

Extra comment coming soon.

2 seconds might also work, yes, but 10 seconds seems more reliable and short enough to never present any real issues. It would actually be harder to reliably test a 2-second delay, because of the round trip time on curl requests; an updated timestamp might mean that the code didn't work, or it might just mean your second request didn't get there fast enough.